modules/tinymce/src/core/test/ts/browser/newline/ForcedRootBlockTest.ts

```diff
@@ -0,0 +1,163 @@
+    });
+  };
+
+  TinyLoader.setupLight(function (editor: Editor, onSuccess, onFailure) {
+    const tinyApis = TinyApis(editor);
+
+    const sAssertNewLine = (label: string, rootBlock: string, initalHTML: string, expectedHTML: (innerHTML: string) => string) => Logger.t(label, GeneralSteps.sequence([
+      Logger.t('Insert block before', GeneralSteps.sequence([
+        tinyApis.sSetContent(initalHTML),
+        tinyApis.sSetCursor([0, 0], 0),
+        sInsertNewline(editor, {}),
+        tinyApis.sAssertContentStructure(
+          ApproxStructure.build((s, str, arr) => {
+            return s.element('body', {
+              children: [
+                ApproxStructure.fromHtml(expectedHTML('<br data-mce-bogus="1">')),
+                s.element(rootBlock, {})
+              ]
+            });
+          })
+        ),
+        tinyApis.sAssertSelection([1, 0], 0, [1, 0], 0)
+      ])),
+      Logger.t('Split block in the middle', GeneralSteps.sequence([
+        tinyApis.sSetContent(initalHTML),
+        tinyApis.sSetCursor([0, 0], 1),
+        sInsertNewline(editor, {}),
+        tinyApis.sAssertContentStructure(
+          ApproxStructure.build((s, str, arr) => {
+            return s.element('body', {
+              children: [
+                s.element(rootBlock, {}),
+                ApproxStructure.fromHtml(expectedHTML('b'))
+              ]
+            });
+          })
+        ),
+        tinyApis.sAssertSelection([1, 0], 0, [1, 0], 0)
+      ])),
+      Logger.t('Insert block after', GeneralSteps.sequence([
+        tinyApis.sSetContent(initalHTML),
+        tinyApis.sSetCursor([0, 0], 2),
+        sInsertNewline(editor, {}),
+        tinyApis.sAssertContentStructure(
+          ApproxStructure.build((s, str, arr) => {
+            return s.element('body', {
+              children: [
+                s.element(rootBlock, {}),
+                ApproxStructure.fromHtml(expectedHTML('<br data-mce-bogus="1">')),
+              ]
+            });
+          })
+        ),
+        tinyApis.sAssertSelection([1], 0, [1], 0)
+      ])),
+      Logger.t('Insert block after bookmark', GeneralSteps.sequence([
+        tinyApis.sSetRawContent(`<${rootBlock}>${bookmarkSpan}<br data-mce-bogus="1"></${rootBlock}>`),
+        tinyApis.sSetCursor([0], 1),
+        sInsertNewline(editor, {}),
+        tinyApis.sAssertContentStructure(
+          ApproxStructure.build((s, str) => {
+            return s.element('body', {
+              children: [
+                s.element(rootBlock, {
+                  children: [
+                    ApproxStructure.fromHtml(bookmarkSpan),
+                    s.element('br', {
+                      attrs: {
+                        'data-mce-bogus': str.is('1')
+                      }
+                    })
+                  ]
+                }),
+                s.element(rootBlock, {
+                  attrs: Obj.map(editor.settings.forced_root_block_attrs, (val) => str.is(val)),
```

Comment:
This probably isn't a good idea to do this, as it's not properly asserting it's what you'd expect. As something else could change this during execution, or something similar. We should instead pass this through to the `sAssertNewLine` function so as it's a constant and can't be changed.

- [ ] Don't rely on a mutable editor variable in assertions